jujugui/static/gui/src/app/app.js

```diff
@@ -599,6 +599,8 @@ YUI.add('juju-gui', function(Y) {
         if (!evt.newVal) {
           // The model is not connected, do nothing waiting for a reconnection.
           console.log('model disconnected');
+          // Need to re-render the provider logo so that it clears.
+          this._renderProviderLogo();
```

Comment:
This should probably be called every time this `connectedChange` is called and then maybe you can remove it from the `_renderComponents` method.